var/spack/repos/builtin/packages/atlas/package.py

```diff
@@ -92,10 +95,8 @@ def install(self, spec, prefix):
 
         # Lapack resource to provide full lapack build. Note that
         # ATLAS only provides a few LAPACK routines natively.
-        lapack_stage = self.stage[1]
-        lapack_tarfile = os.path.basename(lapack_stage.fetcher.url)
-        lapack_tarfile_path = join_path(lapack_stage.path, lapack_tarfile)
-        options.append('--with-netlib-lapack-tarfile=%s' % lapack_tarfile_path)
+        options.append('--with-netlib-lapack-tarfile=%s' %
+                       self.stage[1].archive_file)
```

Comment:
@alalazo @davydden: This isn't directly related to this particular PR, but I am not a huge fan of the stage indexing syntax.  If the user adds more resources, the indexes have to change.  Would it be possible to modify the composite class so that we can refer to resource stages by name?  Then this becomes `self.stage['lapack']`, which I think is more clear and more resilient to package changes.